libres/lib/include/ert/analysis/update.hpp

```diff
@@ -17,5 +17,15 @@ bool smoother_update(std::vector<int> step_list,
                      ensemble_config_type *ensemble_config,
                      enkf_fs_type *source_fs, enkf_fs_type *target_fs,
                      FILE *log_stream, bool verbose);
+void save_parameters(enkf_fs_type *target_fs,
```

Comment:
I think this is not the way to go, this is changes the accessability of the function within this namespace. is this what we want to do to the code.
A beter suggestion is to use `#define private public` where the tests are being writen. for more discussion on the topic please look at https://github.com/catchorg/Catch2/issues/258#issuecomment-37450543

Reply:
Not sure if it is possible to test the function without having it defined in the header file. Any thoughts @pinkwah ?

Reply:
You can add the declaration in the test source file. Now if that is desired or not??

My personal gut reaction is "No - we don't test private functions; we test the public api". But of course in this particular code the amount of testing & design is "a bit limited", and all new testing like this is certainly desirable. If we decide to test functions which have been private I personally think the "honest" way to do it is to add the declaration in the header as suggested hear.

Reply:
@pinkwah helped me with the declaration stuff